src/plugins/data/public/ui/dataset_selector/configurator.tsx

```diff
@@ -52,25 +53,49 @@ export const Configurator = ({
       defaultMessage: "I don't want to use the time filter",
     }
   );
-  const [language, setLanguage] = useState<string>(() => {
+  const [selectedLanguage, setSelectedLanguage] = useState<string>('');
+  const indexedViewsService = type?.indexedViewsService;
+  const [selectedIndexedView, setSelectedIndexedView] = useState<string | undefined>();
+  const [indexedViews, setIndexedViews] = useState<IndexedView[]>([]);
+  const [loadingIndexedViews, setLoadingIndexedViews] = useState(false);
+
+  useEffect(() => {
+    setLanguages(type?.supportedLanguages(dataset) || []);
+  }, [dataset, type]);
+
+  useEffect(() => {
     const currentLanguage = queryString.getQuery().language;
     if (languages.includes(currentLanguage)) {
-      return currentLanguage;
+      setSelectedLanguage(currentLanguage);
     }
```

Comment:
don't you need to return, otherwise, don't you always set languages[0] below?

Reply:
Fixed, thanks for catching